torch/utils/data/dataloader.py

```diff
@@ -631,6 +575,9 @@ def __init__(self, loader: DataLoader) -> None:
         self._drop_last = loader.drop_last
         self._index_sampler = loader._index_sampler
         self._num_workers = loader.num_workers
+        ws, rank = _get_distributed_settings()
```

Comment:
Change 1: Move distributed information to the constructor of `DataLoaderIterator`.